server/tests/src/test/java/org/infinispan/server/persistence/PersistenceIT.java

```diff
@@ -21,16 +39,64 @@
 })
 public class PersistenceIT {
 
-   public static final DatabaseServerListener DATABASE = new DatabaseServerListener("h2", "mysql", "postgres");
+   private static final String DATABASES = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_TYPES);
+   private static final String DATABASE_LIBS = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_LIBS);
+   private static final Integer ARTIFACT_ID_POSITION = 1;
+   private static final String EXTERNAL_JDBC_DRIVER = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_EXTERNAL_DRIVERS);
+   private static String jdbcDriverFromFile = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_DRIVERS_FILE);
+
+   public static final DatabaseServerListener DATABASE_LISTENER = new DatabaseServerListener(getDatabases());
```

Comment:
Previously this defauled to `"h2", "mysql", "postgres"`, however if `INFINISPAN_TEST_CONTAINER_DATABASE_TYPES` is not configured it now defaults to only "h2".

IMO we should make it so that if `INFINISPAN_TEST_CONTAINER_DATABASE_TYPES` is not defined, only "h2" is used, otherwise we should only use the DBs configured via `INFINISPAN_TEST_CONTAINER_DATABASE_TYPES`. E.g. If I configured "mysql,postgres" I don't expect h2 to be included as well.

We then need to ensure that `INFINISPAN_TEST_CONTAINER_DATABASE_TYPES="h2,mysql,postgres"` is configued in the default CI run so that we have the same coverage as before.

Reply:
@ryanemerson If you configured "mysql,postgres" only `mysql,postgres` will run, h2 will note be invoked.
if you don't set `INFINISPAN_TEST_CONTAINER_DATABASE_TYPES` only h2 will be used

Reply:
The current code is:

```java
      Set<String> dbCollection = Stream.of("h2").collect(Collectors.toCollection(HashSet::new));

      //Another databases can be added
      if(DATABASES != null) {
         dbCollection.addAll(Arrays.asList(DATABASES.split(",")));
      }
      return dbCollection.toArray(new String[dbCollection.size()]);
```

You always add "h2" to `dbCollection`.

Reply:
The reason ^ code wasn't rsulting in "h2" always being selected was because the results of this method are always ignored. `DatabaseServerListener`'s constructor reloads the property, so this code is redundant.

I have pushed a commit that makes it so that the default DBs are ["h2", "mysql", "postgres"], but this is overridden if the property is set.